common/src/main/scala/org/apache/celeborn/common/util/Utils.scala

```diff
@@ -994,4 +996,64 @@ object Utils extends Logging {
     }
   }
 
+  def checkedDownCast(value: Long): Int = {
+    val downCast = value.toInt
+    if (downCast.toLong != value) {
+      throw new IllegalArgumentException("Cannot downcast long value " + value + " to integer.")
+    }
+    downCast
+  }
+
+  @throws[IOException]
+  def checkFileIntegrity(fileChannel: FileChannel, length: Int): Unit = {
+    val remainingBytes = fileChannel.size - fileChannel.position
+    if (remainingBytes < length) {
+      logError(
+        s"File remaining bytes not not enough, remaining: ${remainingBytes}, wanted: ${length}.")
+      throw new RuntimeException(s"File is corrupted ${fileChannel}")
+    }
+  }
+
+  @throws[IOException]
+  def readBuffer(fileChannel: FileChannel, buffer: ByteBuffer, length: Int): Unit = {
+    checkFileIntegrity(fileChannel, length)
+    buffer.clear
+    buffer.limit(length)
+    while (buffer.hasRemaining) fileChannel.read(buffer)
+    buffer.flip
+  }
+
+  def readBuffer(fileChannel: FileChannel, buffer: ByteBuf, length: Int): Unit = {
+    checkFileIntegrity(fileChannel, length)
+    val tmpBuffer = ByteBuffer.allocate(length)
+    while (tmpBuffer.hasRemaining) fileChannel.read(tmpBuffer)
+    tmpBuffer.flip()
+    buffer.writeBytes(tmpBuffer)
+  }
+
+  @throws[IOException]
+  def readBuffer(
+      fileChannel: FileChannel,
+      header: ByteBuffer,
+      buffer: ByteBuf,
+      headerSize: Int): Int = {
+    readBuffer(fileChannel, header, headerSize)
+    val bufferLength = header.getInt(12)
+    if (bufferLength <= 0 || bufferLength > buffer.capacity) {
+      logError(s"Incorrect buffer header, buffer length: ${bufferLength}.")
+      throw new RuntimeException(s"File ${fileChannel} is corrupted")
```

Comment:
by fileChannel, we can't directly get the filename if log errors. can readbuffer add a parameter filename ?